### PR TITLE
SE-62 | added controller for get history

### PR DESCRIPTION
**Pull request by suyangyu**

### CODE REVIEW

File: test/uk/gov/hmrc/taxfreechildcareapplicationstatus/helpers/TestConstants.scala [hunk 1]
```diff
@@ -25,5 +27,7 @@ object TestConstants {
 
   val testNino: String = new Generator().nextNino.nino
   val testUniqueClaimId: String = f"${Random.nextInt(Math.pow(10, 11).toInt)}%011d"
+  val testOriginatorId: String = UUID.randomUUID().toString
```
Comment (derek-gray) [inline test/uk/gov/hmrc/taxfreechildcareapplicationstatus/helpers/TestConstants.scala]: Isn't the OriginatorId always the same string and not a UUID?

File: app/uk/gov/hmrc/taxfreechildcareapplicationstatus/controllers/GetTfcHistoryController.scala [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+object GetTfcHistoryController {
+
+  val originatorId = "Originator-Id"
+  val correlationId = "CorrelationId"
+
+  val ninoRegex = "^((?!(BG|GB|KN|NK|NT|TN|ZZ)|(D|F|I|Q|U|V)[A-Z]|[A-Z](D|F|I|O|Q|U|V))[A-Z]{2})[0-9]{6}[A-D]$"
+  val ucidRegex = "^\\d{11}$"
+  val originatorRegex = "^(.*)$"
```
Comment (jennygrahamjones) [inline app/uk/gov/hmrc/taxfreechildcareapplicationstatus/controllers/GetTfcHistoryController.scala]: This doesn't match the regex we've been provided: `^(DA2_NSI)$`

File: app/uk/gov/hmrc/taxfreechildcareapplicationstatus/controllers/GetTfcHistoryController.scala [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      case (true, true, true, true) => action
+      case (false, _, _, _) => Future.successful(BadRequest(Json.toJson(InvalidNinoErr)))
+      case (_, false, _, _) => Future.successful(BadRequest(Json.toJson(InvalidUcidErr)))
+      case (_, _, false, _) => Future.successful(BadRequest(Json.toJson(InvalidOriginatorIdErr)))
+      case (_, _, _, false) => Future.successful(BadRequest) // TODO confirm error json once it's defined
+    }
+
+  def getTfcHistory(nino: String, uniqueClaimsId: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      validateRequest(nino, uniqueClaimsId) {
```
Comment (jennygrahamjones) [inline app/uk/gov/hmrc/taxfreechildcareapplicationstatus/controllers/GetTfcHistoryController.scala]: We should supply the controller with an auth connector, and wrap this call with a check for the `PrivilegedApplication` predicate, like: https://github.com/hmrc/real-time-income-information/blob/3226f7160e172ff7196da0f376e53323303accaf/app/controllers/RealTimeIncomeInformationController.scala#L45

File: it/uk/gov/hmrc/taxfreechildcareapplicationstatus/helpers/IntegrationTestConstants.scala [hunk 4]
```diff
@@ -24,5 +26,7 @@ object IntegrationTestConstants {
 
   val testNino: String = new Generator().nextNino.nino
   val testUniqueClaimId: String = f"${Random.nextInt(Math.pow(10, 11).toInt)}%011d"
+  val testOriginatorId: String = UUID.randomUUID().toString
```
Comment (jennygrahamjones) [inline it/uk/gov/hmrc/taxfreechildcareapplicationstatus/helpers/IntegrationTestConstants.scala]: This isn't the format of the originator ID. We could use this for a failure case, but no a success response.

File: test/uk/gov/hmrc/taxfreechildcareapplicationstatus/helpers/TestConstants.scala (hunk 1 above)
Comment (suyangyu) [reply test/uk/gov/hmrc/taxfreechildcareapplicationstatus/helpers/TestConstants.scala]: We're not exposing that in an open repo unless there is no security risks

File: it/uk/gov/hmrc/taxfreechildcareapplicationstatus/helpers/IntegrationTestConstants.scala (hunk 4 above)
Comment (suyangyu) [reply it/uk/gov/hmrc/taxfreechildcareapplicationstatus/helpers/IntegrationTestConstants.scala]: We're not exposing the exact string until we're cleared that it's not a security risk

File: app/uk/gov/hmrc/taxfreechildcareapplicationstatus/controllers/GetTfcHistoryController.scala (hunk 3 above)
Comment (suyangyu) [reply app/uk/gov/hmrc/taxfreechildcareapplicationstatus/controllers/GetTfcHistoryController.scala]: We're not exposing the exact string until we're cleared that it's not a security risk

Comment (jennygrahamjones) [reply app/uk/gov/hmrc/taxfreechildcareapplicationstatus/controllers/GetTfcHistoryController.scala]: This has nothing to do with a String. You just need to wrap the call in `authorised(AuthProviders(PrivilegedApplication)) {` and provide an auth connector.

**Comment by jennygrahamjones** [comment]
LGTM :+1:

**Pull request merged**